employees/views.py

```diff
@@ -49,6 +59,70 @@ def perform_create(self, serializer: ReportSerializer) -> None:
         serializer.save(author=self.request.user)
 
 
+class MonthNavigationMixin(ContextMixin):
+    url_name = ""
+
+    def _get_previous_month_url(self, year: int, month: int, pk: int) -> str:
+        date = datetime.date(year=year, month=month, day=1) + relativedelta(months=-1)
+        return reverse(self.url_name, kwargs={"pk": pk, "year": date.year, "month": date.month})
+
+    def _get_next_month_url(self, year: int, month: int, pk: int) -> str:
+        date = datetime.date(year=year, month=month, day=1) + relativedelta(months=+1)
+        return reverse(self.url_name, kwargs={"pk": pk, "year": date.year, "month": date.month})
+
+    def _get_recent_month_url(self, pk: int) -> str:
+        date = timezone.now().date()
+        return reverse(self.url_name, kwargs={"pk": pk, "year": date.year, "month": date.month})
+
+    @staticmethod
+    def _get_title_date(year: int, month: int) -> str:
+        date = datetime.date(year=year, month=month, day=1)
+        return date.strftime("%m/%y")
+
+    def _get_month_navigator_params(self) -> dict:
+        disable_next_button = False
+        disable_previous_button = False
+        year = int(self.kwargs["year"])
+        month = int(self.kwargs["month"])
+        pk = self.kwargs.get("pk", None)
+
+        if month == MONTH_NAVIGATION_FORM_MAX_MONTH_VALUE and year == MONTH_NAVIGATION_FORM_MAX_YEAR_VALUE:
+            disable_next_button = True
+        elif month == MONTH_NAVIGATION_FORM_MIN_MONTH_VALUE and year == MONTH_NAVIGATION_FORM_MIN_YEAR_VALUE:
+            disable_previous_button = True
+
+        return {
+            "path": self.request.path,
+            "navigation_text": MonthNavigationText,
+            "month_form": MonthSwitchForm(initial_date=datetime.date(year=year, month=month, day=1)),
+            "next_month_url": self._get_next_month_url(year, month, pk),
+            "recent_month_url": self._get_recent_month_url(pk),
+            "previous_month_url": self._get_previous_month_url(year, month, pk),
+            "disable_next_button": disable_next_button,
+            "disable_previous_button": disable_previous_button,
+            "title_date": self._get_title_date(year, month),
+        }
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        context_data.update(self._get_month_navigator_params())
+        return context_data
+
+    def redirect_to_another_month(self, request: HttpRequest) -> HttpResponseRedirectBase:
+        post_data = request.POST.copy()
```

Comment:
If I get this right, this function redirect to another month, if the form is valid. I would suggest changing the logic a bit:
```python
post_data = request.POST
        form = MonthSwitchForm(data=post_data)
        if not form.is_valid():
            return redirect(request.path)

        new_date = datetime.datetime.strptime(post_data["date"], "%m-%Y")
        return redirect(
            reverse(
                self.url_name,
                kwargs={"pk": self.kwargs["pk"], "year": new_date.year, "month": new_date.month},
            )
        )
```

it seems more readable to me and does nt copy anything.

Reply:
The `form.is_valid()` will fail every time, because date in POST is passed as a string and the `MonthSwitchForm` doesn't handle that and needs a date input instead. This is why I switched the date in `request.POST` before passing it to form, but I had to clone it first to have it mutable.